lib/encode_url.js

```diff
@@ -24,7 +24,8 @@ const encodeURL = (str) => {
     if (parsed.origin === 'null') return str;
 
     parsed.search = encodeURI(safeDecodeURI(parsed.search));
-    return parsed.toString();
+    // preserve host (international domain name) as is.
+    return format(parsed, { unicode: true});
```

Comment:
url.format() doesn't support port number in Node 8, see https://github.com/hexojs/hexo-util/pull/114.

Reply:
A workaround is to use punycode api

``` js
const { toUnicode } = require('./punycode');
const url = parsed.toString().replace(parsed.hostname, toUnicode(parsed.hostname);
```

Reply:
readme also needs to be updated.

Reply:
node 8 EOL is arround the corner, and will probably imply that this package, as well as many other and also hexo have a major version bump

Reply:
we can use the workaround temporarily until node 8 EOL. I don't think this is a breaking change, I do agree IDN users would prefer not to punycode.

Reply:
Added the workaround. another change introduced by this PR is that IDN whether it's in unicode or punycoded is always encoded to unicode.

``` js
encodeURL(http://xn--br-mia.com/baz')
// http://bár.com/baz
```